### PR TITLE
[CI] Buildkite upload wheels to S3 and docker

### PR DESCRIPTION
**Pull request by simon-mo**
## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: ci/travis/build-docker-images.py [hunk 1]
```diff
@@ -408,13 +408,17 @@ def push_readmes(merge_build: bool):
...
+            build_type = MERGE
     if build_type == HUMAN:
         _configure_human_version()
-    if build_type in {HUMAN, MERGE, BUILDKITE
-                      } or _check_if_docker_files_modified():
+    if build_type in {HUMAN, MERGE} or _check_if_docker_files_modified():
         DOCKER_CLIENT = docker.from_env()
         is_merge = build_type == MERGE
-        if is_merge:
+        if is_merge:  # Buildkite should authenticate in the background.
```
Comment (ijrsvt) [inline ci/travis/build-docker-images.py:421]: Does the _get_docker_creds fn work for Buildkite?

File: ci/travis/build-docker-images.py [hunk 2]
```diff
@@ -408,13 +408,17 @@ def push_readmes(merge_build: bool):
...
             # We do this here because we want to be authenticated for
             # Docker pulls as well as pushes (to avoid rate-limits).
             username, password = _get_docker_creds()
```
Comment (ijrsvt) [inline ci/travis/build-docker-images.py:424]: If we are already logged in. Do not do this.

File: ci/travis/build-docker-images.py [hunk 3]
```diff
@@ -408,13 +408,19 @@ def push_readmes(merge_build: bool):
...
+            build_type = MERGE
     if build_type == HUMAN:
         _configure_human_version()
-    if build_type in {HUMAN, MERGE, BUILDKITE
-                      } or _check_if_docker_files_modified():
+    if build_type in {HUMAN, MERGE} or _check_if_docker_files_modified():
         DOCKER_CLIENT = docker.from_env()
         is_merge = build_type == MERGE
-        if is_merge:
+        # Buildkite should authenticate in the background.
```
Comment (ijrsvt) [inline ci/travis/build-docker-images.py:422]:
```suggestion
        # Buildkite is authenticated in the background.
```

**Review by ijrsvt — APPROVED** [review]
This looks good! The only thing I would check is that 'logging in within a  subprocess call' works for using those credentials outside of the subprocess.

**Comment by simon-mo** [comment]
Good point, let me push this to a test-wheel branch to see if we can actually push the images...

**Comment by simon-mo** [comment]
Status: Docker seems to be okay. But we need to push wheels to /{branch_name} in addition to /{sha}.

**Comment by simon-mo** [comment]
Wheels seem to be happy. One last sanity check to cross check the travis location: https://buildkite.com/ray-project/ray-builders-branch/builds/1650#b2b29015-a294-4d28-b6d6-c2a100b7dede/1071-3992

**Pull request merged**